src/Recomposer.tsx

```diff
@@ -192,11 +194,11 @@ export class Recomposer<
     test: predicate<InnerProps>,
     trueEnhancer:
       | ComponentEnhancer<InnerProps, NextProps>
-      | InferableComponentEnhancer<NextProps>
+      | InferableComponentEnhancer<{}>
```

Comment:
I think it would be ideal to get `recompose` to allow `InferableComponentEnhancer<NextProps>` type option in its exported `.d.ts`.